### ISSUE TITLE
Is v3 officially released?

### ISSUE THREAD

**Issue by artamonovkirill** [actions/setup-java#295]
**Description:**
I got a PR from Renovate bot bumping the `setup-java` action to v3.

**Task version:**
v3:
```
     - uses: actions/setup-java@v3
        with:
          java-version: 17
          distribution: adopt
          cache: gradle
```

**Platform:**
- [x] Ubuntu
- [ ] macOS
- [ ] Windows

**Runner type:**
- [x] Hosted
- [ ] Self-hosted

**Repro steps:**  
https://github.com/artamonovkirill/ninshubur/pull/189

**Expected behavior:**
CI step succeeds

**Actual behavior:**
CI steps fails due to out-of-range error:
```
Run actions/setup-java@v3
  with:
    java-version: 17
    distribution: adopt
    cache: gradle
    java-package: jdk
    architecture: x64
    check-latest: false
    server-id: github
    server-username: GITHUB_ACTOR
    server-password: GITHUB_TOKEN
    overwrite-settings: true
    job-status: success
Trying to resolve the latest version from remote
Resolved latest version as 17.0.[2](https://github.com/artamonovkirill/ninshubur/runs/5359803125?check_suite_focus=true#step:6:2)+8
Trying to download...
Downloading Java 17.0.2+8 (Adopt-Hotspot) from https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.2%2B8/OpenJDK17U-jdk_x64_linux_hotspot_17.0.2_8.tar.gz ...
Extracting Java archive...
/usr/bin/tar xz --warning=no-unknown-keyword -C /home/runner/work/_temp/db9051e8-02a1-4d70-a166-91e[3](https://github.com/artamonovkirill/ninshubur/runs/5359803125?check_suite_focus=true#step:6:3)325db[4](https://github.com/artamonovkirill/ninshubur/runs/5359803125?check_suite_focus=true#step:6:4)[5](https://github.com/artamonovkirill/ninshubur/runs/5359803125?check_suite_focus=true#step:6:5)0 -f /home/runner/work/_temp/03d793fc-d48f-4a35-ba8c-beb8[6](https://github.com/artamonovkirill/ninshubur/runs/5359803125?check_suite_focus=true#step:6:6)95bfd11
Java 1[7](https://github.com/artamonovkirill/ninshubur/runs/5359803125?check_suite_focus=true#step:6:7).0.2+[8](https://github.com/artamonovkirill/ninshubur/runs/5359803125?check_suite_focus=true#step:6:8) was downloaded
Setting Java 17.0.2+8 as the default

Java configuration:
  Distribution: adopt
  Version: 17.0.2+8
  Path: /opt/hostedtoolcache/Java_Adopt_jdk/17.0.2-8/x64

Creating settings.xml with server-id: github
Writing to /home/runner/.m2/settings.xml
Received 58720256 of 2[9](https://github.com/artamonovkirill/ninshubur/runs/5359803125?check_suite_focus=true#step:6:9)35425672 (2.0%), 56.0 MBs/sec
Received 167772160 of 2935425672 (5.7%), 80.0 MBs/sec
Received 327155712 of 2935425672 (11.1%), [10](https://github.com/artamonovkirill/ninshubur/runs/5359803125?check_suite_focus=true#step:6:10)3.7 MBs/sec
Received 528482304 of 2935425672 (18.0%), 125.7 MBs/sec
Received 717225984 of 2935425672 (24.4%), 136.5 MBs/sec
Received 910163968 of 2935425672 (31.0%), 144.4 MBs/sec
Received 1098907648 of 2935425672 (37.4%), 149.4 MBs/sec
Received 1283457024 of 2935425672 (43.7%), 152.7 MBs/sec
Received 1430257664 of 2935425672 (48.7%), 151.3 MBs/sec
Received 1598029824 of 2935425672 (54.4%), 152.2 MBs/sec
Received 1757413376 of 2935425672 (59.9%), 152.2 MBs/sec
Received 1895825408 of 2935425672 (64.6%), 150.5 MBs/sec
Received 2046820352 of 2935425672 (69.7%), 150.0 MBs/sec
Received 2172649472 of 2935425672 (74.0%), 147.8 MBs/sec
Received 2294284288 of 2935425672 (78.2%), 145.7 MBs/sec
Received 2420[11](https://github.com/artamonovkirill/ninshubur/runs/5359803125?check_suite_focus=true#step:6:11)3408 of 2935425672 (82.4%), [14](https://github.com/artamonovkirill/ninshubur/runs/5359803125?check_suite_focus=true#step:6:14)4.1 MBs/sec
Received 2545942528 of 2935425672 (86.7%), 142.7 MBs/sec
Received 2684354560 of 2935425672 (91.4%), 142.1 MBs/sec
Received 2831[15](https://github.com/artamonovkirill/ninshubur/runs/5359803125?check_suite_focus=true#step:6:15)5[20](https://github.com/artamonovkirill/ninshubur/runs/5359803125?check_suite_focus=true#step:6:20)0 of 2935425672 (96.4%), 141.9 MBs/sec
Received 2935425672 of 2935425672 (100.0%), 140.6 MBs/sec
Error: The value of "length" is out of range. It must be >= 0 && <= [21](https://github.com/artamonovkirill/ninshubur/runs/5359803125?check_suite_focus=true#step:6:21)47483647. Received 29354[25](https://github.com/artamonovkirill/ninshubur/runs/5359803125?check_suite_focus=true#step:6:25)672
```

**Comment by ChrisCarini** [comment]
I, too, am experiencing a similar issue. [example execution link](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true)

Snippet of task output:
```
Run actions/setup-java@f69f00b5e53[2](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:2)4696b07f6b1c92f0470a6df00780
  with:
    distribution: zulu
    java-version: 11
    cache: gradle
    java-package: jdk
    architecture: x64
    check-latest: false
    server-id: github
    server-username: GITHUB_ACTOR
    server-password: GITHUB_TOKEN
    overwrite-settings: true
    job-status: success
Trying to resolve the latest version from remote
Resolved latest version as 11.0.14+9
Trying to download...
Downloading Java 11.0.14+9 (Zulu) from https://cdn.azul.com/zulu/bin/zulu11.54.2[3](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:3)-ca-jdk11.0.1[4](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:4)-linux_x64.tar.gz ...
Extracting Java archive...
/usr/bin/tar xz --warning=no-unknown-keyword -C /home/runner/work/_temp/[5](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:5)bec91df-a7[6](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:6)8-4c41-bff2-f09233f6ee10 -f /home/runner/work/_temp/f3[7](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:7)004f6-715f-409d-9[8](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:8)82-8b4661[9](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:9)1bd8b
Java 11.0.14+9 was downloaded
Setting Java 11.0.14+9 as the default

Java configuration:
  Distribution: zulu
  Version: 11.0.14+9
  Path: /opt/hostedtoolcache/Java_Zulu_jdk/11.0.14-9/x64

Creating settings.xml with server-id: github
Writing to /home/runner/.m2/settings.xml
Received 0 of 2734770113 (0.0%), 0.0 MBs/sec
Received 83886080 of 2734770113 (3.1%), 40.0 MBs/sec
Received 209715200 of 2734770113 (7.7%), 66.6 MBs/sec
Received 322961408 of 2734770113 (11.8%), 76.9 MBs/sec
Received 452984832 of 2734770113 (16.6%), 86.3 MBs/sec
Received 562036736 of 2734770113 (20.6%), 89.3 MBs/sec
Received 662700032 of 2734770113 (24.2%), 90.2 MBs/sec
Received 784334848 of 2734770113 (28.7%), 93.4 MBs/sec
Received 89758[10](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:10)56 of 2734770[11](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:11)3 (32.8%), 95.1 MBs/sec
Received 1006632960 of 2734770113 (36.8%), 96.0 MBs/sec
Received 1[12](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:12)4073472 of 27347701[13](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:13) (41.1%), 97.4 MBs/sec
Received 1228931072 of 2734770113 (44.9%), 97.6 MBs/sec
Received 1337982976 of 2734770113 (48.9%), 98.1 MBs/sec
Received [14](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:14)47034880 of 2734770113 (52.9%), 98.5 MBs/sec
Received [15](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:15)60281088 of 2734770113 (57.1%), 99.1 MBs/sec
Received [16](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:16)73527296 of 2734770113 (61.2%), 99.7 MBs/sec
Received [17](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:17)82579200 of 2734770113 (65.2%), 99.9 MBs/sec
Received [18](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:18)79048[19](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:19)2 of 2734770113 (68.7%), 99.5 MBs/sec
Received 1979711488 of 2734770113 (72.4%), 99.3 MBs/sec
Received 2101346304 of 2734770113 (76.8%), 100.1 MBs/sec
Received 2214592512 of 2734770113 (81.0%), 100.5 MBs/sec
Received 233[20](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:20)33024 of 2734770113 (85.3%), 101.0 MBs/sec
Received 2441084928 of 2734770113 (89.3%), 101.1 MBs/sec
Received 2541748224 of 2734770113 (92.9%), 100.9 MBs/sec
Received 2642411520 of 2734770113 (96.6%), 100.7 MBs/sec
Received 2713798593 of 2734770113 (99.2%), 99.4 MBs/sec
Received 2734770113 of 2734770113 (100.0%), 99.2 MBs/sec
Error: The value of "length" is out of range. It must be >= 0 && <= [21](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:21)47483647. Received [27](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:27)[34](https://github.com/ChrisCarini/environment-variable-settings-summary-intellij-plugin/runs/5630609896?check_suite_focus=true#step:3:34)770113
```

**Comment by attilaolah** [comment]
I'm seeing a similar problem, except it gives me a warning instead of an error. So subsequent steps can still complete, but are very slow since apparently there is no cache.

**Comment by dmitry-shibanov** [comment]
Hello everyone. We prepared a fix and updated the major tag with these changes. Could you please confirm that everything works as expected ?

**Comment by artamonovkirill** [comment]
The setup step succeeds now!
```
Run actions/setup-java@v3
Trying to resolve the latest version from remote
Resolved latest version as 17.0.2+8
Trying to download...
Downloading Java 17.0.2+8 (Adopt-Hotspot) from https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.2%2B8/OpenJDK17U-jdk_x64_linux_hotspot_17.0.2_8.tar.gz ...
Extracting Java archive...
/usr/bin/tar xz --warning=no-unknown-keyword -C /home/runner/work/_temp/757b372f-37bf-4f27-93d4-5928ea1d7cf8 -f /home/runner/work/_temp/dfe4c63f-fc30-4755-b2f6-30aa06690bf8
Java 17.0.2+8 was downloaded
Setting Java 17.0.2+8 as the default

Java configuration:
  Distribution: adopt
  Version: 17.0.2+8
  Path: /opt/hostedtoolcache/Java_Adopt_jdk/17.0.2-8/x64

Creating settings.xml with server-id: github
Writing to /home/runner/.m2/settings.xml
Received 234881024 of 2935425672 (8.0%), 221.3 MBs/sec
Received 528482304 of 2935425672 (18.0%), 249.4 MBs/sec
Received 792723456 of 2935425672 (27.0%), 247.6 MBs/sec
Received 1056964608 of 2935425672 (36.0%), 248.7 MBs/sec
Received 1346371584 of 2935425672 (45.9%), 254.0 MBs/sec
Received 1631584256 of 2935425672 (55.6%), 256.5 MBs/sec
Received 1895825408 of 2935425672 (64.6%), 255.8 MBs/sec
Received 2[14](https://github.com/artamonovkirill/ninshubur/runs/5793264375?check_suite_focus=true#step:6:14)7483647 of 2935425672 (73.2%), 135.0 MBs/sec
Received 2302672895 of 2935425672 (78.4%), 135.8 MBs/sec
Received 2436890623 of 2935425672 (83.0%), 135.4 MBs/sec
Received 2566914047 of 2935425672 (87.4%), 134.7 MBs/sec
Received 2713714687 of 2935425672 (92.4%), 135.0 MBs/sec
Received 28479324[15](https://github.com/artamonovkirill/ninshubur/runs/5793264375?check_suite_focus=true#step:6:15) of 2935425672 (97.0%), 134.7 MBs/sec
Received 2935425672 of 2935425672 (100.0%), 119.3 MBs/sec
Cache Size: ~2799 MB (2935425672 B)
/usr/bin/tar --use-compress-program zstd -d -xf /home/runner/work/_temp/c7552e25-fb8b-46f7-866a-b5002cf1cb3d/cache.tzst -P -C /home/runner/work/ninshubur/ninshubur
Cache restored successfully
Cache restored from key: setup-java-Linux-gradle-6ced1418278f67deda07f1d033aa9a4486e3c[16](https://github.com/artamonovkirill/ninshubur/runs/5793264375?check_suite_focus=true#step:6:16)573ab59694[37](https://github.com/artamonovkirill/ninshubur/runs/5793264375?check_suite_focus=true#step:6:37)f0ff61e0f79e4
```
